### PR TITLE
Included controller ID in the location object returned by the available resources API

### PR DESCRIPTION
**Pull request by dbozzato81**

### CODE REVIEW

File: v1/pkg/client/api/openapi.yaml [hunk 1]
```diff
@@ -3522,11 +3522,13 @@ components:
...
+            ControllerID: 046b6c7f-0b8a-43b9-b35d-6489e6daee91
             Region: Central
             ID: 046b6c7f-0b8a-43b9-b35d-6489e6daee91
             PrivateNetworkVIDSupport: true
           - DataCenter: FTC1
             PrivateNetworkVNISupport: true
+            ControllerID: 046b6c7f-0b8a-43b9-b35d-6489e6daee91
             Region: Central
             ID: 046b6c7f-0b8a-43b9-b35d-6489e6daee91
             PrivateNetworkVIDSupport: true
```
Comment (neilgierman) [inline v1/pkg/client/api/openapi.yaml:3534]: I know you are just following what is already there, but shouldn't each location in the locations list have a different DataCenter name, ID, and now ControllerID? I don't think it is a real world example to have 2 locations exactly the same. This same comment applies in many places in this file.

File: v1/pkg/client/api/openapi.yaml [hunk 2]
```diff
@@ -5118,7 +5131,12 @@ components:
           description: Indicates whether or not the fabric dedicated for private networks
             supports VNI ID.
           type: boolean
+        ControllerID:
+          description: Unique ID for the controller that manages the data center.
+          format: uuid
+          type: string
       required:
+      - ControllerID
```
Comment (neilgierman) [inline v1/pkg/client/api/openapi.yaml:5139]: What if we are early in Day -1 where we have a Hoster (Tenant) and Pod (Site) in the DB but has not gone to factory yet, so we have no controller attached?

File: v1/pkg/client/api/openapi.yaml (hunk 1 above)
Comment (dbozzato81) [reply v1/pkg/client/api/openapi.yaml:3534]: Yes it is not a real example but the file is generated.

File: v1/pkg/client/api/openapi.yaml (hunk 2 above)
Comment (dbozzato81) [reply v1/pkg/client/api/openapi.yaml:5139]: It will return empty string, we started to use required after the "omitempty" discussion with GUI team.

**Review by neilgierman — APPROVED** [review]

**Comment by mchuang3** [comment]
@neilgierman or @dbozzato81, can you please add some notes to the PR text on the reasons for adding this so there's some history for tracking purposes? Thanks.

**Pull request merged**